app/src/main/java/com/jvoye/tasky/core/data/auth/EncryptedSessionDataStore.kt

```diff
@@ -0,0 +1,38 @@
+package com.jvoye.tasky.core.data.auth
+
+import android.content.Context
+import androidx.datastore.dataStore
+import com.jvoye.tasky.core.domain.AuthInfo
+import com.jvoye.tasky.core.domain.SessionStorage
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.withContext
+import kotlin.random.Random
+
+private val Context.dataStore by dataStore(
+    fileName = "api-token-storage",
+    serializer = AuthInfoSerializer
+)
+
+class EncryptedSessionDataStore(
+    val context: Context
+): SessionStorage {
+
+    override suspend fun get(): AuthInfo {
+        return withContext(Dispatchers.IO) {
+            context.dataStore.data.first().toAuthInfo()
+        }
```

Comment:
This `withContext` is not necessary here, since you're not wrapping any blocking calls (just suspending ones). It won't cause any issues or performance problems, but I'm a friend of just switching when necessary, since otherwise I've noticed that people quickly start to put these `withContext` blocks everywhere in the codebase and then it starts to become a mess